diff -up SHTOOLS-3.1/src/PlmBar.f95.openmp SHTOOLS-3.1/src/PlmBar.f95
--- SHTOOLS-3.1/src/PlmBar.f95.openmp	2015-07-21 01:59:49.000000000 -0600
+++ SHTOOLS-3.1/src/PlmBar.f95	2016-05-18 14:15:21.799767710 -0600
@@ -120,6 +120,7 @@ subroutine PlmBar(p, lmax, z, csphase, c
             
     scalef = 1.0d-280
     
+!$omp critical (PlmBar_init)
     if (lmax > lmax_old) then
         if (allocated (sqr)) deallocate (sqr)
         if (allocated (f1)) deallocate (f1)
@@ -175,6 +176,7 @@ subroutine PlmBar(p, lmax, z, csphase, c
         lmax_old = lmax
     
     end if
+!$omp end critical (PlmBar_init)
     
     !---------------------------------------------------------------------------
     !   
